notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   "outputs": [],
+   "source": [
+    "y_pred = clf_Ca.predict(X_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here are some scoring metrics for our first pass at this model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.mean_squared_error(y_test,y_pred)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.r2_score(y_test,y_pred)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Feature importances and selection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "There are ways to save time without giving up model performance.  One way is by using GradientBoostingRegressor's **feature_importances** attribute, which can identify the most important features in determining a split, allowing us to cast out less important features."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below we print out the highest ranking features\n",
+    "(remember, these are the numbered labels of our Principal Components):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "importances=clf_Ca.feature_importances_\n",
+    "indices = np.argsort(importances)[::-1]\n",
+    "\n",
+    "#Print the feature ranking\n",
+    "print(\"Feature ranking:\")\n",
+    "\n",
+    "#for f in range(X.shape[1]):\n",
+    "for f in range(10):\n",
+    "    print(\"{:d}.  feature {:d} ({:f})\".format(f+1,indices[f],importances[indices[f]]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the most part, the important features appear to also correspond to the the PCs that explain the most variance.  \n",
+    "However, one shouldn't read too much into the feature importances - \n",
+    "a low rank does not necessarily mean that the feature isn't important in determining outputs, \n",
+    "because there may be high correlation between features.  \n",
+    "\n",
+    "For more information, you can read [this post](http://alexperrier.github.io/jekyll/update/2015/08/27/feature-importance-random-forests-gini-accuracy.html) \n",
+    "describing feature engineering and selection in random forests - \n",
+    "most insights will apply to GBRTs as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(15,5))\n",
+    "plt.bar(range(len(X_train)),\n",
+    "        clf_Ca.feature_importances_[indices[range(len(X_train))]]/clf_Ca.feature_importances_[indices[0]])\n",
+    "plt.xlabel('feature rank')\n",
+    "plt.ylabel('feature importance (normalized to max)')"
```

Comment:
Add a `;` to the end to suppress the output
`<matplotlib.text.Text at 0x1091a9f28>`